### DISCUSSION TITLE
lsp is not working if a test file is put outside the lazyvim config folder

### DISCUSSION THREAD

**Discussion by hopezh** [LazyVim/LazyVim#2572] [Q&A]
As shown below, using `K` to show `hover` info is working for a test file put within the `lazyvim` config folder, but not when it's moved to another folder.

The main difference I can found is related to the `root directory` of `lsp info` as shown below. 

This happened a few days ago, and I'm not sure if it's related to the various updates in lazyvim.  

May I ask what could be causing this issue, and how to fix it? 

Image 1: lsp is working for a test python file when it's put inside the lazyvim config folder:
<img width="1441" alt="Screenshot 2024-02-18 at 23 09 57" src="https://github.com/LazyVim/LazyVim/assets/23435810/d2e2b3de-7101-42aa-85a6-bf2f0feb2b62">

Image 2: `pyright` lsp is attached in this case, the `root directory` is "**_/User/oatmini/.config/lazyvim_**": 
<img width="1441" alt="Screenshot 2024-02-18 at 23 40 04" src="https://github.com/LazyVim/LazyVim/assets/23435810/a551b7d1-0a79-4f8c-a807-b72fd8f57b6b">

Image 3: lsp is not working, i.e. using `K` for hover info on line 5 shows nothing, when the same file is put outside the lazyvim config folder:
<img width="1441" alt="Screenshot 2024-02-18 at 23 11 09" src="https://github.com/LazyVim/LazyVim/assets/23435810/0a7ec8a5-44fc-46ee-97d8-83c35115cf04">

Image 4: The `pyright` lsp is also attached, but the `root directory` is "**_/User/oatmini_**" only, which is not where the current file is stored: 
<img width="1441" alt="Screenshot 2024-02-18 at 23 11 31" src="https://github.com/LazyVim/LazyVim/assets/23435810/839511df-bbcb-4eea-a2e1-83160d56d68c">

**Comment by dpetka2001** [comment]
It doesn't matter where you put the file. `pyright` looks for [these patterns](https://github.com/neovim/nvim-lspconfig/blob/d1bab4cf4b69e49d6058028fd933d8ef5e74e680/lua/lspconfig/server_configurations/pyright.lua#L3-L11) to detect the `root_dir` or else it's running in `single file mode`, so that shouldn't be a problem.
![2024-02-18_18-17](https://github.com/LazyVim/LazyVim/assets/12776461/6b4e88d3-30a4-4b29-8de4-ae9e79eeb498)
![2024-02-18_18-18](https://github.com/LazyVim/LazyVim/assets/12776461/515cf0cf-a124-41f4-ab18-9890c27d2621)
